### PR TITLE
SQL: Switch server side for geosql from ShapeBuilder to libs/geo

### PR DESCRIPTION
**Pull request by imotov**
Switches the server side of the geosql processing from using
ShapeBuilder to libs/geo geometry objects.

Relates to #29872

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java [hunk 1]
```diff
@@ -41,41 +48,49 @@
...
-        shapeBuilder = new PointBuilder(lon, lat);
+        shape = new Point(lat, lon);
     }
 
     public GeoShape(Object value) throws IOException {
-        shapeBuilder = ShapeParser.parse(value);
+        try {
+            shape = parse(value);
+        } catch (ParseException ex) {
+            throw new ElasticsearchParseException("cannot load shape", ex);
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java]: Why do you need to wrap it in a new exception here? And if it's the case shouldn't this be an `Sql` exception?
Also can we have some tests to check those cases?

**Review by matriv — COMMENTED** [review]
Looks good, left a comment regarding exception handling during parsing.

**Review by astefan — APPROVED** [review]
LGTM, but I do agree with @matriv's questions and request.

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java (hunk 1 above)
Comment (imotov) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java]: > Why do you need to wrap it in a new exception here? And if it's the case shouldn't this be an Sql exception?

I wrapped it here for consistency with the next constructor which cannot throw anything besides IOException, I will rewrap the it into `SqlIllegalArgumentException` that sounds like a better idea.

> Also can we have some tests to check those cases?

We are already testing this cases in [`StWkttosqlProcessorTests`](https://github.com/elastic/elasticsearch/blob/08da5bb5eadd0efd79ba5fad325da29219b89623/x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/StWkttosqlProcessorTests.java#L35).

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java [hunk 2]
```diff
@@ -58,15 +57,15 @@ public GeoShape(Object value) throws IOException {
...
+            throw new SqlIllegalArgumentException("Cannot parse [" + value + "] as a geo_shape value", ex);
         }
     }
 
     public GeoShape(StreamInput in) throws IOException {
         try {
             shape = parse(in.readString());
         } catch (ParseException ex) {
-            throw new ElasticsearchParseException("cannot load shape", ex);
+            throw new SqlIllegalArgumentException("cannot load geo_shape", ex);
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java]: why not adding the read value here too? (save in.readString() to a var)

**Review by matriv — APPROVED** [review]
LGTM

**Comment by imotov** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**